### PR TITLE
[3494] Add auditing to allocations

### PR DESCRIPTION
**Pull request by dankmitchell**
🚢

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Add auditing to allocations](https://trello.com/c/mZG5eeNK/3494-add-auditing-to-allocations)

File: app/models/allocation.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ class Allocation < ApplicationRecord
   belongs_to :accredited_body, class_name: "Provider"
   belongs_to :recruitment_cycle
 
+  audited
```
Comment (misaka) [inline app/models/allocation.rb:6]: Shouldn't we associate this with the provider?

Comment (misaka) [reply app/models/allocation.rb:6]:
```suggestion
  audited associated_with: :provider
```

File: spec/models/allocation_spec.rb [hunk 2]
```diff
@@ -6,6 +6,10 @@
       subject.valid?
     end
 
+    describe "auditing" do
+      it { should be_audited }
```
Comment (misaka) [inline spec/models/allocation_spec.rb:10]:
```suggestion
      it { should be_audited.associated_with(:provider) }
```

File: app/models/allocation.rb (hunk 1 above)
Comment (asmega) [reply app/models/allocation.rb:6]: I checked earlier
```
=> #<Audited::Audit:0x00007fea288ac6d0
 id: 259515,
 auditable_id: 436,
 auditable_type: "Allocation",
 associated_id: nil,
 associated_type: nil,
 user_id: 8916,
 user_type: "User",
 username: nil,
 action: "update",
 audited_changes: {"request_type"=>[1, 2], "number_of_places"=>[28, 0]},
 version: 2,
 comment: nil,
 remote_address: "::1",
 request_uuid: "3c498ddc-5048-4214-a819-28cebe7370fd",
 created_at: Thu, 04 Jun 2020 10:07:41 UTC +00:00>
```
The user and allocation is sufficient as that associates with everything that we need

**Pull request merged**